service/src/main/java/de/starwit/mapper/FieldMapper.java

```diff
@@ -0,0 +1,150 @@
+package de.starwit.mapper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import de.starwit.dto.FieldDto;
+import de.starwit.dto.FieldType;
+import de.starwit.dto.FieldValidateRulesType;
+import de.starwit.persistence.entity.Attribute;
+import de.starwit.persistence.entity.DataType;
+
+@Component
+public class FieldMapper implements CustomMapper<Attribute, FieldDto> {
+
+    @Override
+    public FieldDto convertToDto(Attribute entity) {
+        FieldDto dto = new FieldDto();
+        dto.setId(entity.getId());
+        dto.setFieldName(entity.getName());
+        dto.setFieldType(FieldType.valueOf(entity.getDataType().toString()));
+        setValidationToDto(dto, entity);
+        return dto;
+    }
+
+    @Override
+    public Attribute convertToEntity(FieldDto dto) {
+        Attribute entity = new Attribute();
+        entity.setId(dto.getId());
+        entity.setName(dto.getFieldName());
+        if (dto.getFieldType() != null) {
+            entity.setDataType(DataType.valueOf(dto.getFieldType().toString()));
+        }
+        setValidationToEntity(dto, entity);
+        return entity;
+    }
+
+    private void setValidationToEntity(FieldDto dto, Attribute entity){
+        if (dto == null || dto.getFieldType() == null || dto.getFieldValidateRules() == null) {
+            return;
```

Comment:
Is an exception better suited here? Could this be potentially hazardous, when this is null?